docs/guidance/rayservice.md

```diff
@@ -109,149 +121,158 @@ rayservice-sample-serve-svc                        ClusterIP   10.100.39.92
-
-### RayService Observability
-You can use `kubectl logs` to check the operator logs or the head/worker nodes logs.
-You can also use `kubectl describe rayservices rayservice-sample` to check the states and event logs of your RayService instance.
-
-For Ray Serve monitoring, you can refer to the [Ray observability documentation](https://docs.ray.io/en/master/ray-observability/state/state-api.html).
-To run Ray state APIs, log in to the head pod by running `kubectl exec -it <head-node-pod> bash` and use the Ray CLI or you can run commands locally using `kubectl exec -it <head-node-pod> -- <ray state api>`.
-
-For example, `kubectl exec -it <head-node-pod> -- ray summary tasks` outputs the following:
-```shell
-======== Tasks Summary: 2022-07-28 15:10:24.801670 ========
-Stats:
-------------------------------------
-total_actor_scheduled: 17
-total_actor_tasks: 5
-total_tasks: 0
-
-
-Table (group by func_name):
-------------------------------------
-    FUNC_OR_CLASS_NAME                 STATE_COUNTS    TYPE
-0   ServeController.listen_for_change  RUNNING: 5      ACTOR_TASK
-1   ServeReplica:MangoStand.__init__   FINISHED: 3     ACTOR_CREATION_TASK
-2   HTTPProxyActor.__init__            FINISHED: 2     ACTOR_CREATION_TASK
-3   ServeReplica:PearStand.__init__    FINISHED: 3     ACTOR_CREATION_TASK
-4   ServeReplica:OrangeStand.__init__  FINISHED: 3     ACTOR_CREATION_TASK
-5   ServeReplica:FruitMarket.__init__  FINISHED: 3     ACTOR_CREATION_TASK
-6   ServeReplica:DAGDriver.__init__    FINISHED: 2     ACTOR_CREATION_TASK
-7   ServeController.__init__           FINISHED: 1     ACTOR_CREATION_TASK
-```
+In Step 7, modifying `serveConfigV2` will not trigger a zero downtime upgrade for Ray clusters.
+Instead, it will reapply the new configurations to the existing RayCluster.
+However, if you modify `spec.rayClusterConfig` in the RayService YAML file, it will trigger a zero downtime upgrade for Ray clusters.
+To elaborate, RayService will temporarily create a new RayCluster and wait for it to be ready, then switch traffic to the new RayCluster by updating the selector of the head service managed by RayService (i.e. `rayservice-sample-head-svc`) and terminate the old one.
 
-### Delete the RayService Instance
-```
-$ kubectl delete -f config/samples/ray_v1alpha1_rayservice.yaml
-```
+To elaborate, during the zero downtime upgrade process, RayService will create a new RayCluster temporarily and wait for it to become ready. 
+Once the new RayCluster is ready, RayService will update the selector of the head service managed by RayService (i.e., `rayservice-sample-head-svc`) to point to the new RayCluster to switch the traffic to the new RayCluster. 
+Finally, the old RayCluster will be terminated.
+
+There are certain exceptions that will not trigger a zero downtime upgrade.
+Currently, only the fields managed by Ray Autoscaler, such as `replicas` and `scaleStrategy.workersToDelete`, will not trigger a zero downtime upgrade.
```

Comment:
Ideally we can write down the entire list of such fields.  Are there too many to list, or are we worried that they'll frequently change?

Reply:
`replicas` and `scaleStrategy.workersToDelete` are the only two exceptions at this moment.

Reply:
I will merge this PR and cut a release branch. If you have any further comments, we can address them later.